src/Mod/PartDesign/App/FeaturePipe.cpp

```diff
@@ -272,31 +325,38 @@ App::DocumentObjectExecReturn *Pipe::execute(void)
                 TopTools_ListOfShape sim;
                 mkPS.Simulate(2, sim);
 
+                // Note that while we call them front and back, these sections
+                // appear to correspond to the front or back of the path. When one
+                // or both ends of the pipe are points, one or both of these wires
+                // (and eventually faces) will be null.
                 frontwires.push_back(TopoDS::Wire(sim.First()));
                 backwires.push_back(TopoDS::Wire(sim.Last()));
             }
         }
 
         BRepBuilderAPI_MakeSolid mkSolid;
 
-        if (!frontwires.empty()) {
-            // build the end faces, sew the shell and build the final solid
-            TopoDS_Shape front = Part::FaceMakerCheese::makeFace(frontwires);
-            TopoDS_Shape back  = Part::FaceMakerCheese::makeFace(backwires);
-
+        if (!frontwires.empty() || !backwires.empty()) {
             BRepBuilderAPI_Sewing sewer;
             sewer.SetTolerance(Precision::Confusion());
-            sewer.Add(front);
-            sewer.Add(back);
 
-            for (TopoDS_Shape& s : shells)
+            // build the end faces, sew the shell and build the final solid
+            if (!frontwires.empty()) {
+                TopoDS_Shape front = Part::FaceMakerCheese::makeFace(frontwires);
+                sewer.Add(front);
+            }
+            if (!backwires.empty()) {
+                TopoDS_Shape back  = Part::FaceMakerCheese::makeFace(backwires);
+                sewer.Add(back);
+            }
+            for (auto& s : shells)
                 sewer.Add(s);
 
             sewer.Perform();
             mkSolid.Add(TopoDS::Shell(sewer.SewedShape()));
         } else {
             // shells are already closed - add them directly
-            for (TopoDS_Shape& s : shells) {
+            for (auto& s : shells) {

```

Comment:
Why this change? the explicit type name helps others to understand the code. Therefore as long as the type name is there, we can keep it.

Reply:
I think this one can stay. The for loop already says that `s` runs through every element of `shells`.

Reply:
This is nit-picking, I know but once the explicit type name is there, just keep it.

Reply:
If you insist. I'll revert this change and force push shortly.